src/core/ext/transport/chaotic_good/server_transport.cc

```diff
@@ -0,0 +1,332 @@
+    frame.stream_id = stream_id;
+    return Map(outgoing_frames.Send(std::move(frame)),
+               [](bool success) -> absl::Status {
+                 if (!success) {
+                   // Failed to send outgoing frame.
+                   return absl::UnavailableError("Transport closed.");
+                 }
+                 return absl::OkStatus();
+               });
+  };
+  return Seq(
+      TrySeq(
+          // Wait for initial metadata then send it out.
+          call_initiator.PullServerInitialMetadata(),
+          [send_fragment](ServerMetadataHandle md) mutable {
+            ServerFragmentFrame frame;
+            frame.headers = std::move(md);
+            return send_fragment(std::move(frame));
+          },
+          // Continuously send client frame with client to server messages.
+          ForEach(OutgoingMessages(call_initiator),
+                  [send_fragment, aligned_bytes = aligned_bytes_](
+                      MessageHandle message) mutable {
+                    ServerFragmentFrame frame;
+                    // Construct frame header (flags, header_length and
+                    // trailer_length will be added in serialization).
+                    const uint32_t message_length =
+                        message->payload()->Length();
+                    const uint32_t padding =
+                        message_length % aligned_bytes == 0
+                            ? 0
+                            : aligned_bytes - message_length % aligned_bytes;
+                    GPR_ASSERT((message_length + padding) % aligned_bytes == 0);
+                    frame.message = FragmentMessage(std::move(message), padding,
+                                                    message_length);
+                    return send_fragment(std::move(frame));
+                  })),
+      call_initiator.PullServerTrailingMetadata(),
+      [send_fragment](ServerMetadataHandle md) mutable {
+        ServerFragmentFrame frame;
+        frame.trailers = std::move(md);
+        return send_fragment(std::move(frame));
+      });
+}
+
+auto ChaoticGoodServerTransport::DeserializeAndPushFragmentToNewCall(
+    FrameHeader frame_header, BufferPair buffers) {
+  ClientFragmentFrame fragment_frame;
+  ScopedArenaPtr arena(acceptor_->CreateArena());
+  absl::Status status = transport_.DeserializeFrame(
+      frame_header, std::move(buffers), arena.get(), fragment_frame);
+  absl::optional<CallInitiator> call_initiator;
+  if (status.ok()) {
+    auto create_call_result =
+        acceptor_->CreateCall(*fragment_frame.headers, arena.release());
+    if (create_call_result.ok()) {
+      call_initiator.emplace(std::move(*create_call_result));
+      call_initiator->SpawnGuarded(
+          "server-write", [this, stream_id = frame_header.stream_id,
+                           call_initiator = *call_initiator]() {
+            return CallOutboundLoop(stream_id, call_initiator);
+          });
```

Comment:
Should the call_initiator be stored into stream_map_ here ?

Reply:
Yeah... it'll get fixed in the next PR.